### PR TITLE
Factor out reactExamples module from previous demo module

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
We had a few LK React page / webpart examples in the `demo` module which we referenced from the ui-components docs and have directed client developers to. The `demo` module also used the @labkey/build package and was used as an example of how to use that to generated LK React endpoints. With the removal of the `demo` module, we are moving all of those resources to their own `reactExamples` module.

#### Related Pull Requests
* https://github.com/LabKey/tutorialModules/pull/156

#### Changes
* factor out reactExamples module from previous demo module
* update page URL controllers and doc reference links

### CODE REVIEW

File: reactExamples/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+start the server, from the command line in the reactExamples module directory run `npm start`, then navigate to either of the appropriate development
+pages (note the "Dev" added to the end of the action name).
+- http://localhost:8080/labkey/home/reactExamples-helloWorldDev.view
+- http://localhost:8080/labkey/home/reactExamples-queryModelDev.view
+- http://localhost:8080/labkey/home/reactExamples-todoListDev.view
+- http://localhost:8080/labkey/home/reactExamples-fileAttachmentFormDev.view
+
+<a name="jest"></a>
+## Jest Tests
+This module is setup to run Jest tests, including using Enzyme and Jest Snapshot testing. There is an example Jest test and
```
Comment (labkey-susanh) [inline reactExamples/README.md:50]: Is it worth converting to React Testing Library here?

**Review by labkey-susanh — APPROVED** [review]

File: reactExamples/README.md (hunk 1 above)
Comment (cnathe) [reply reactExamples/README.md:50]: Good idea! Done.

Comment (labkey-susanh) [reply reactExamples/README.md:50]: Probably also want to update this text too. :-)

Comment (cnathe) [reply reactExamples/README.md:50]: yes....I just merged. I'll open a new PR shortly with this updated.

**Pull request merged**